### PR TITLE
Added HAProxy ASG between external NLB and internal ALB

### PR DESCRIPTION
**Pull request by marcus-bcl**
https://dsdmoj.atlassian.net/jira/software/projects/DAM/boards/340

Tested in sandpit - confirmed the app is now only accessible from whitelisted IPs and has static EIPs on the front.

From my whitelisted IP:
```
$ curl -I https://ndelius.sandpit.delius-core.probation.hmpps.dsd.io/NDelius-war/delius/JSP/healthcheck.jsp
HTTP/2 200 

$ host ndelius.sandpit.delius-core.probation.hmpps.dsd.io
ndelius.sandpit.delius-core.probation.hmpps.dsd.io is an alias for dlc-sandpit-ndelius-ext-22da161992370e22.elb.eu-west-2.amazonaws.com.
dlc-sandpit-ndelius-ext-22da161992370e22.elb.eu-west-2.amazonaws.com has address 52.56.105.62
dlc-sandpit-ndelius-ext-22da161992370e22.elb.eu-west-2.amazonaws.com has address 35.177.161.207
dlc-sandpit-ndelius-ext-22da161992370e22.elb.eu-west-2.amazonaws.com has address 35.176.77.223
```

From external un-whitelisted IP:
Screenshot 2019-08-29 at 14 11 20

### CODE REVIEW

File: security-groups/weblogic-interface.tf [hunk 1]
```diff
@@ -80,6 +80,26 @@ resource "aws_security_group_rule" "interface_external_elb_egress_umt" {
   description              = "Out to UMT instances"
 }
 
+resource "aws_security_group_rule" "interface_lb_self_ingress" {
```
Comment (crimesp) [inline security-groups/weblogic-interface.tf]: do LBs need to have access to themselves?

File: security-groups/weblogic-ndelius.tf [hunk 2]
```diff
@@ -80,6 +80,26 @@ resource "aws_security_group_rule" "ndelius_public_subnet_ingress_tls" {
   description       = "Public subnet in (TLS)"
 }
 
+resource "aws_security_group_rule" "ndelius_lb_self_ingress" {
+  security_group_id         = "${aws_security_group.weblogic_ndelius_lb.id}"
+  type                      = "ingress"
+  protocol                  = "tcp"
+  from_port                 = "80"
+  to_port                   = "80"
+  self                      = true
```
Comment (crimesp) [inline security-groups/weblogic-ndelius.tf]: do LBs need to have access to themselves?

Comment (marcus-bcl) [reply security-groups/weblogic-ndelius.tf]: this is to support inter-LB communication, we have the same SGs applied to the HAProxy LB and the ALB so this allows them to talk to each other

Comment (crimesp) [reply security-groups/weblogic-ndelius.tf]: ahh ok that makes sense now!

**Pull request merged**